aws/signer/v4/v4.go

```diff
@@ -311,6 +311,42 @@ func (v4 Signer) Presign(r *http.Request, body io.ReadSeeker, service, region st
 	return v4.signWithBody(r, body, service, region, exp, true, signTime)
 }
 
+// SignString creates a signature of provided string according to AWS v4
+// signing process described in the documentation:
+// https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html
+//
+// It may be used to create signatures of "Browser-Based Upload using HTTP POST"
+// policy documents to be used in the browser. In which case, make sure to
+// base64 encode the policy document using base64.StdEncoding before creating
+// it's signature.
+func (v4 Signer) SignString(awsContext aws.Context, stringToSign string, service, region string, exp time.Duration, signTime time.Time) (string, error) {
+	currentTimeFn := v4.currentTimeFn
+	if currentTimeFn == nil {
+		currentTimeFn = time.Now
+	}
+
+	ctx := &signingCtx{
+		Time:                   signTime,
+		ExpireTime:             exp,
+		ServiceName:            service,
+		Region:                 region,
+		DisableURIPathEscaping: v4.DisableURIPathEscaping,
+		unsignedPayload:        v4.UnsignedPayload,
```

Comment:
Minor, but it's probably worth not initializing `signingCtx` with these two values. Reason being is that they go unused since you are invoking `ctx.buildSignature()` directly. I think it would alleviate any future confusion on their intent as it relates to this function.